src/composables/transactionItemList.ts

```diff
@@ -0,0 +1,145 @@
+  token: string,
+}
+
+type ITransactionElement = ITransaction | IActiveMultisigTx
+
+export function useTransactionList({
+  store,
+  isMultisig,
+  token,
+}: TransactionItemList) {
+  const availableTokens = computed<ITokenList>(() => store.state.fungibleTokens.availableTokens);
+  const transactions = computed(() => store.state.transactions);
+  const getTxSymbol = computed(() => store.getters.getTxSymbol);
+  const getTxDirection = computed(() => store.getters.getTxDirection);
+  const getDexContracts = computed<IDexContracts>(() => store.getters.getDexContracts);
+
+  const getAccountPendingTransactions = computed<IActiveMultisigTx[]>(
+    () => store.getters.getAccountPendingTransactions,
+  );
+
+  const canLoadMore = computed(() => !!transactions.value.nextPageUrl);
+
+  const {
+    searchPhrase,
+    displayMode,
+    FILTER_MODE,
+  } = useTransactionAndTokenFilter();
+
+  const { pendingMultisigTransaction } = usePendingMultisigTransaction({ store });
+
+  function isFungibleTokenTx(tx: ITx) {
+    return Object.keys(availableTokens.value).includes(tx.contractId);
+  }
+
+  function filterProperTransactions(transactionList: ITransactionElement[]) {
+    return transactionList.filter((transaction) => {
+      if (!token) {
+        return true;
+      }
+
+      const innerTx = getInnerTransaction(transaction.tx!);
+      if (token !== AETERNITY_CONTRACT_ID) {
+        return innerTx?.contractId === token;
+      }
+
+      return !innerTx.contractId || !isFungibleTokenTx(innerTx);
+    });
+  }
+
+  function filterTransactionByDisplayMode(transactionList: ITransactionElement[]) {
+    return transactionList.filter((transaction) => {
+      const direction = getTxDirection.value(transaction.tx);
+      const isDex = isTxDex(transaction.tx!, getDexContracts.value);
+
+      switch (displayMode.value.key) {
+        case FILTER_MODE.all:
+          return true;
+        case FILTER_MODE.dex:
+          return isDex;
+        case FILTER_MODE.out:
+          return direction === TX_DIRECTION.sent && !isDex;
+        case FILTER_MODE.in:
+          return direction === TX_DIRECTION.received;
+        default:
+          throw new Error(`${i18n.t('pages.recentTransactions.unknownMode')} ${displayMode.value.key}`);
+      }
+    });
+  }
+
+  function filterTransactionByPhrase(transactionList: ITransactionElement[]) {
+    return transactionList.filter(
+      (transaction) => !searchPhrase.value || getTxSymbol.value(transaction)
+        .toLocaleLowerCase()
+        .includes(searchPhrase.value.toLocaleLowerCase()),
+    );
+  }
+
+  const loadedTransactionList = computed((): ITransactionElement[] => {
+    const transactionList: ITransactionElement[] = [
+      ...getAccountPendingTransactions.value,
+      ...(transactions.value as any).loaded as ITransaction[],
+    ];
+
+    if (isMultisig && pendingMultisigTransaction.value?.tx) {
+      transactionList.push(pendingMultisigTransaction.value);
+    }
+
+    return transactionList;
```

Comment:
Can be written in a shorter form:
```suggestion
    return [
      ...getAccountPendingTransactions.value,
      ...(transactions.value as any).loaded as ITransaction[],
      ...((isMultisig && pendingMultisigTransaction.value?.tx)
        ? pendingMultisigTransaction.value
        : []
      ),
    ];
```